### PR TITLE
Improve petset e2es

### PR DESCRIPTION
**Pull request by bprashanth**
Mainly
- scale petset to 0 and wait for volume provisioner to delete pvs so we don't leak them
- bake binaries into init[0] and cp into destination, instead of wgetting at runtime
- dump verbose debug output on failure

https://github.com/kubernetes/test-infra/pull/115, https://github.com/kubernetes/kubernetes/issues/26824

### CODE REVIEW

File: test/e2e/petset.go [hunk 1]
```diff
@@ -51,6 +52,9 @@ const (
 	redisManifestPath       = "test/e2e/testing-manifests/petset/redis"
 )
 
+// Time: 25m, slow by design.
+// GCE Quota requirements: 3 pds, one per pet manifest declared above.
+// GCE Api requirements: nodes and master need storage r/w permissions.
 var _ = framework.KubeDescribe("PetSet", func() {
```
Comment (jlowdermilk) [inline test/e2e/petset.go]: Add "[Slow]" to description, since this is slow by design.

**Comment by bprashanth** [comment]
@jlowdermilk PTAL I added a new commit to explicitly join new members to zookeeper master, if there is one

**Comment by jlowdermilk** [comment]
Still lgtm, but please squash commits.

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #26834

**Pull request merged**